### PR TITLE
feat: install Grafana with Prometheus datasource

### PR DESCRIPTION
**Pull request by xcoulon**
- create a ServiceAccount with `cluster-monitoring-view` role
- retrieve the ServiceAccount's token and use it as a bearer token
to connect to Thanos Querier (which is itself "protected with
an openshift/oauth-proxy sidecar container)

Fixes https://issues.redhat.com/browse/CRT-809

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: config/monitoring/cluster_monitoring_config.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: openshift-monitoring
```
Comment (alexeykazakov) [inline config/monitoring/cluster_monitoring_config.yaml:4]: It's going to be a problem in OSD.
I don't think we can't create anything in openshift-* namespaces in OSD.
So, I guess user workload monitoring is not enableable in OSD? Worth checking with app-sre and/or platform-sre folks.

File: config/monitoring/user_workload_monitoring_config.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: user-workload-monitoring-config
+  namespace: openshift-user-workload-monitoring
```
Comment (alexeykazakov) [inline config/monitoring/user_workload_monitoring_config.yaml:5]: Same here.

File: config/oauth/idp.yaml [hunk 3]
```diff
@@ -5,7 +5,7 @@ metadata:
   name: cluster
 spec:
   identityProviders:
-  - mappingMethod: lookup
+  - mappingMethod: ${MAPPING}
```
Comment (alexeykazakov) [inline config/oauth/idp.yaml:8]: Are you parameterizing it for local testing? We should never use anything but "lookup" in our stage and prod env.
If it's for local development then fine. But let's mention it in readme.

File: config/monitoring/cluster_monitoring_config.yaml (hunk 1 above)
Comment (xcoulon) [reply config/monitoring/cluster_monitoring_config.yaml:4]: yeah, we need to check that indeed 🤯

File: config/oauth/idp.yaml (hunk 3 above)
Comment (xcoulon) [reply config/oauth/idp.yaml:8]: yes, the default value remains `lookup` as you can see in https://github.com/codeready-toolchain/toolchain-infra/pull/34/files#diff-3c99f4ffee6524f96786b5c5a37a00a16a3a5c7ba1b44975e251578a31a3a632R14-R17
and yes, I wanted to use the `claim` mapping method for local testing. I'll mention that in the README ;)

Comment (xcoulon) [reply config/oauth/idp.yaml:8]: done in ad952ea

**Comment by MatousJobanek** [comment]
@xcoulon I guess that we can close this PR, right?

**Comment by xcoulon** [comment]
ah yes, indeed! Sorry for the late response 😳

**Pull request closed without merging**